Tools/machines/lxplus-cern/spack.yaml

```
@@ -0,0 +1,47 @@
+spack:
+# CUDA
+  definitions:
+  - cuda: []
+  - cuda: [cuda ^ncurses+symlinks, blaspp +cuda ^ncurses+symlinks]
+    when: env.get("SPACK_STACK_USE_CUDA", "") == "1"
+
+# basic Python dependencies for WarpX Python PICMI bindings
+  - python: []
+  - python: [python ^ncurses+symlinks, py-pip ^ncurses+symlinks, py-mpi4py ^ncurses+symlinks, py-numpy ^ncurses+symlinks,
+      py-periodictable ^ncurses+symlinks, py-picmistandard ^ncurses+symlinks]
+    when: env.get("SPACK_STACK_USE_PYTHON", "") == "1"
+  specs:
+  - ncurses+symlinks
+  - adios2 ^ncurses+symlinks
+  - blaspp ^ncurses+symlinks
+  - ccache ^ncurses+symlinks
+  - cmake ^ncurses+symlinks
+  - fftw ^ncurses+symlinks
+  - hdf5 ^ncurses+symlinks
+  - lapackpp ^ncurses+symlinks
+  - mpi ^ncurses+symlinks
+  - openpmd-api ^ncurses+symlinks
+  - pkgconfig
+  - $cuda
+  - $python
+
+  packages:
+    all:
+      target: [x86_64]
+      compiler: [gcc@9.2.0]
+  compilers:
```

On this cluster, is there a pre-installed MPI that we should use?
In that case, we would register it here like this:
```suggestion
    openmpi:
      externals:
      - spec: "openmpi"
        prefix: /cvmfs/sft.cern.ch/lcg/releases/openmpi/lives-here/maybe/edit-me
      buildable: False
  compilers:
```

---

So far I didn't rely on the pre-installed packages (in theory we also have CMake and Cuda for example). At this point I think it makes sense trying to as many of them as possible in order to reduce the time needed to install all the dependencies